job.py

```
@@ -389,89 +390,137 @@
+        celery_task = AsyncResult(task_id, app=celery_app) # Re-initialize celery_task here
+        try:
+
+            while True:
+
+                try:                
+                    # Yield heartbeat every 30 seconds to keep connection alive
+                    # if retries > 0 and retries % 30 == 0:
+                    #     yield b"data: {\"type\":\"heartbeat\"}\n"
+
+                    # messages = await redis_xread(redis, {channel: '0'}, count=None, block=10000)
+                    if retries > max_retries and not completed_yielded or (celery_task.ready() and retries > 3):
```

Ruff (0.13.1)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)